### PR TITLE
Solve https://github.com/galaxyproject/tools-iuc/issues/1742

### PR DESCRIPTION
**Pull request by lldelisle**
I implemented the `--outWigType` option and all derivatives, both in STAR and STAR solo.

### CODE REVIEW

File: tools/rgrnastar/macros.xml [hunk 1]
```diff
@@ -280,4 +280,90 @@
             <action name="column_names" type="metadata" default="GeneID,Counts_unstrand,Counts_firstStrand,Counts_secondStrand" />
         </actions>
     </xml>
+    <xml name="outWig">
+        <section name="coverage" title="Coverage" expanded="false">
```
Comment (bernt-matthias) [inline tools/rgrnastar/macros.xml]: Sections should contain >1 parameters. Here its only one (conditional). I think we can remove the section ..

File: tools/rgrnastar/rg_rnaStar.xml [hunk 2]
```diff
@@ -973,6 +980,107 @@ used: >=5 mappings => MAPQ=0; 3-4 mappings => MAPQ=1; 2 mappings => MAPQ=3. This
...
+                    <param name="outWigTypeSecondWord" value="read2"/>
+                    <param name="outWigStrand" value="false" />
+                </conditional>
+            </section>
+            <output name="output_log" file="rnastar_test.log" compare="re_match_multiline" />
+            <output name="splice_junctions" file="rnastar_test_splicejunctions_PE.bed"/>
+            <output name="mapped_reads" file="rnastar_test_mapped_reads_PE.bam" compare="sim_size" delta="634" />
+            <output name="reads_per_gene" file="tophat_test_reads_per_gene_PE.txt" />
+            <output name="signal_unique_str1" file="tophat_Signal.Unique.both.read2.out.wig" />
+            <output name="signal_uniquemultiple_str1" file="tophat_Signal.Unique.both.read2.out.wig" />
```
Comment (bernt-matthias) [inline tools/rgrnastar/rg_rnaStar.xml]: Can you check for the ftype of the outputs in order to check if setting it works.

File: tools/rgrnastar/rg_rnaStarSolo.xml [hunk 3]
```diff
@@ -101,7 +101,7 @@
     --quantMode TranscriptomeSAM $solo.quantModeGene
     #set $tag_names = str($solo.outSAMattributes).replace(',', ' ')
     --outSAMattributes $tag_names
-    #if "CB" in $tag_names or "UB" in $tag_names:
+    #if "CB" in $tag_names or "UB" in $tag_names or str($coverage.outWig.outWigType) != 'None':
         --outSAMtype BAM SortedByCoordinate
     #else:
         --outSAMtype BAM Unsorted
```
Comment (bernt-matthias) [inline tools/rgrnastar/rg_rnaStarSolo.xml:107]: The format of the bam output should be set to `unsorted.bam` in this case, or?

**Review by bernt-matthias — COMMENTED** [review]
Few comments from my side

File: tools/rgrnastar/macros.xml (hunk 1 above)
Comment (lldelisle) [reply tools/rgrnastar/macros.xml]: You are right. I thought it was a good way to hide it but I can remove it.

File: tools/rgrnastar/rg_rnaStar.xml (hunk 2 above)
Comment (lldelisle) [reply tools/rgrnastar/rg_rnaStar.xml]: Good point

File: tools/rgrnastar/rg_rnaStarSolo.xml (hunk 3 above)
Comment (lldelisle) [reply tools/rgrnastar/rg_rnaStarSolo.xml:107]: No below there are samtools command that sort it.

**Comment by lldelisle** [comment]
@bernt-matthias Thanks for the review I will incorporate these changes tonight.

File: tools/rgrnastar/rg_rnaStarSolo.xml (hunk 3 above)
Comment (lldelisle) [reply tools/rgrnastar/rg_rnaStarSolo.xml:107]: Here is the line: https://github.com/lldelisle/tools-iuc/blob/9c5210c19fda711a9fb9af1cc2aed425a3c04512/tools/rgrnastar/rg_rnaStarSolo.xml#L150

**Review by bernt-matthias — APPROVED** [review]
good to go

**Comment by lldelisle** [comment]
Cool, thank you very much

**Pull request merged**